interfaces/hotplug/udevadm.go

```diff
@@ -81,62 +81,46 @@ func scanDoubleNewline(data []byte, atEOF bool) (advance int, token []byte, err
 	return 0, nil, nil
 }
 
-func parseUdevadmOutput(cmd *exec.Cmd, rd *bufio.Scanner, devices chan<- *HotplugDeviceInfo, parseErrors chan<- error) {
-	defer close(devices)
-	defer close(parseErrors)
-
+func parseUdevadmOutput(cmd *exec.Cmd, rd *bufio.Scanner) (devices []*HotplugDeviceInfo, parseErrors []error) {
 	for rd.Scan() {
 		block := rd.Text()
 		env, err := parseEnvBlock(block)
 		if err != nil {
-			parseErrors <- err
+			parseErrors = append(parseErrors, err)
 		} else {
-			outputDevice(env, devices, parseErrors)
+			dev, err := NewHotplugDeviceInfo(env)
+			if err != nil {
+				parseErrors = append(parseErrors, err)
+			} else {
+				devices = append(devices, dev)
+			}
 		}
 	}
 
 	if err := rd.Err(); err != nil {
-		parseErrors <- fmt.Errorf("failed to read udevadm output: %s", err)
+		parseErrors = append(parseErrors, fmt.Errorf("cannot read udevadm output: %s", err))
 	}
 	if err := cmd.Wait(); err != nil {
-		parseErrors <- fmt.Errorf("failed to read udevadm output: %s", err)
+		parseErrors = append(parseErrors, fmt.Errorf("cannot to read udevadm output: %s", err))
```

Comment:
Please change `cannot to read...` to `cannot read...`

Reply:
Uh, thanks for spotting, fixed.